### DISCUSSION TITLE
Create Artifact Container - Attempt X of 5 failed with error: read ECONNRESET

### DISCUSSION THREAD

**Discussion by serene-capybara** [community/community#76352] [Actions]
### Select Topic Area

Bug

### Body

Hi all,

At some point between Oct 5th and Oct 10th, my pipelines that use the artifact download/upload actions started failing with this error: `Create Artifact Container - Attempt X of 5 failed with error: read ECONNRESET`

I use a self-hosted runner set, deployed with the official Helm chart. Nothing has changed about my network configuration or runner configuration since the last successful run. The runners have no issues picking up jobs or executing workflows, with the exception of uploading and downloading artifacts.

My suspicion is that this is a localized failure of the artifacts API. The regional server [ group] that my pipeline attempts to connect with for the artifact actions is `pipelinesghubeus15.actions.githubusercontent.com`.
The workflow always hangs when trying to access `https://pipelinesghubeus15.actions.githubusercontent.com/r8dyTuVBCaBBvrF4ZabhEJAUz4aT9C4fGTYWOfjTHZEAxZ5tjr/_apis/pipelines/workflows/6697319442/artifacts?api-version=6.0-preview`

I'm wondering if any other users have experienced this and/or if there is a way to confirm or rule-out my suspicion.

Best

**Comment by ghost** [comment]
We see the same issue on self-hosted runners with `pipelinesghubeus24.actions.githubusercontent.com` since yesterday. Do you still have this problem @serene-capybara ?
```
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Artifact name is valid!
##[debug]Artifact Url: https://pipelinesghubeus24.actions.githubusercontent.com/dl[...]xV/_apis/pipelines/workflows/6[...]7/artifacts?api-version=6.0-preview
Create Artifact Container - Attempt 1 of 5 failed with error: read ECONNRESET
```

**Reply by serene-capybara** [reply]
Hi @f-maroszek, sorry for the delayed response. Yes, my workflows involving the artifact API have been affected since October. I retry every day I'm in the office and it still seems to hang on the same upload/download steps.

**Reply by ghost** [reply]
Thanks for the answer, we fixed it now internally - problem was related to MTU misconfiguration in our docker network, maybe it will help you. From internal discussions:
`The MTU was again too big. So within the docker network we have 1500 and that results in the TCP syn using 1460 on GitHub's side. But layers above it (CNI etc) have 1450. So no response is getting through to the dev container.`

**Reply by serene-capybara** [reply]
@f-maroszek, thank you for sharing - this is likely what I'm experiencing as well. The default network interface of the dind sidecar has an MTU of 1450, which is lower than `docker0` (1500).

May I ask what was your solution? I've tried to set the MTU by various methods but no luck so far.

**Reply by ghost** [reply]
All I know is that the solution was iptable rule which rewrites the TCP packets, but I'm not sure about the details - @Wielewout , can you comment?

**Reply by Wielewout** [reply]
The iptable rule put in place is the following one:

```sh
iptables -I FORWARD -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1410
```

This was only meant as a work-around to be able to move forward though.

I haven't found a way though to configure the runner and/or dind container to use an mtu of max 1450 (the one on the main interfaces of the servers and cni).

**Reply by serene-capybara** [reply]
Much appreciated, @Wielewout, @f-maroszek ; I'll give this a try. 

In my case, the affected workflows were those that had steps that ran in a container. My lay-man workaround was to simply avoid containerized steps and live with the overhead of installing dependencies as part of the workflow.

Hopefully there will soon be a more permanent solution upstream for propagating network configuration. I'm not sure if that's on Action Runner's plate or Docker's.

Anyway, thanks again for helping me identify the root cause here!
